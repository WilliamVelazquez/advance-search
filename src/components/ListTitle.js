import React from 'react';
import styled from 'styled-components';

const Title = styled.h2`
  margin: 0;
  color: ${(props) => props.color};
  font-size: 24px;
  position: relative;
  padding-bottom: 2px;
  &:after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    width: 40%;
    height: 4px;
    background: ${(props) => props.lineColor};
    border-radius: 10px;
  }
  @media (max-width: 768px) {
    font-size: 20px;
  }
`;

const ListTitle = (props) => {
  const { title = '', color = '#3c75ff', lineColor = '#3c75ff' } = props;
  return <Title color={color} lineColor={lineColor}>{title}</Title>;
};

export default ListTitle;
