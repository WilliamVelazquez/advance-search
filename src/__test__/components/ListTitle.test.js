import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import 'jest-styled-components';

import ListTitle from '../../components/ListTitle';

describe('<ListTitle />', () => {
  const listTitle = mount(<ListTitle />);
  test('Rendering ListTitle Component', () => {
    expect(listTitle.length).toEqual(1);
  });
});

describe('ListTitle Snapshot', () => {
  test('Verifying ListTitle Component UI', () => {
    const listTitle = create(<ListTitle />);
    expect(listTitle.toJSON()).toMatchSnapshot();
  });
});
