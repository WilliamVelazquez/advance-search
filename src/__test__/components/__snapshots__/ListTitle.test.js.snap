// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTitle Snapshot Verifying ListTitle Component UI 1`] = `
.c0 {
  margin: 0;
  color: #3c75ff;
  font-size: 24px;
  position: relative;
  padding-bottom: 2px;
}

.c0:after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  width: 40%;
  height: 4px;
  background: #3c75ff;
  border-radius: 10px;
}

@media (max-width:768px) {
  .c0 {
    font-size: 20px;
  }
}

<h2
  className="c0"
  color="#3c75ff"
>
  
</h2>
`;
